### PR TITLE
cds_sync: errors display fixes

### LINKED ISSUE #604: CDS -> ILS Synchronisation - Library feedback
The synchronisation works well and meets our needs. 
It is fine running the script ever day with the modifications of the latest 24h. 

Here are few modifications to be done: 
- [x] Ignore 562__ $$c
- [x] Ignore 693__ $$s
- [x] Error messages to be clarified str: [UNEXPECTED INPUT VALUE] Ex. https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/108 -> line 4
- [x] Error messages to be clarified str: None is not of type 'string' Ex. https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/108 -> line 6
- [x] Previous imports, show the latest 20 rows and not the 10 latest rows (in case we cannot check the synchro everyday, and have other imports in the meantime...)
- [x] Creation of new e-items for new records: could we use the same rules as during the migration? If not possible, select by default Not Open Access.
- [x] For the 1st synchro on production side, we need to get the modified records since the migration (21/04/2021), since all the ILSYNC records have been updated since then, we need to be able to check the logs in a file (it will be easier than checking 80 webpages...).

### LINKED ISSUE #606: Importer - Library feedback
**Files used for the tests**
[files.zip](https://github.com/CERNDocumentServer/cds-ils/files/7235522/files.zip)

- SPR-exemple1-add.xml
 -> https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/120 (Preview) and https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/121 (Import)
- SPR-exemple2-add.xml -> https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/119 (Preview)
- EBL-exemple1-add.xml -> https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/109 (Preview) and https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/110 (Import)
- SAF-exemple-add.xml -> https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/123 (preview) https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/124 (Import)
- SAF-exemple-del.xml -> https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/125 (preview) and https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/127 (import)
- EBL-exemple-del-2-ok.xml https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/115 (preview)  and https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/118 (import)

**Name of the providers**
In the importer, could we name the providers as follow: 
- SAFARI => O'reilly (Safari)
- EBL => EbookCentral (EBL)

**General comments**
- [ ] BUG When 'Ignore missing import rules' is ticked, it cannot be un-ticked.
- [ ] FEATURE On the 'Previous import' history, could display the name of the uploaded file. It would help in case we have to refer to the corresponding file.
- [ ] FEATURE On the Import page, could display the name of the uploaded file and the import date/time. 
- [ ] BUG Error is not always explained and therefore difficult to understand: e.g. line 188: https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/109
~~BUG (vizualisation) https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/109 : on page 5 in the preview, there are no more the navigation to the other pages at the bottom, it seems the page gets bigger, and we need to scroll right to find it)~~ comment: couldn't reproduce
~~BUG https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/110 : cannot navigate after page 4. (same as above)~~ comment: couldn't reproduce
- [ ] BUG Discrepancies between Preview and Import:
 ~~Line  4 looks ok in the preview (no error): https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/109
but when importing, there is an error: https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/110
Line  3 looks ok in the preview (no error): https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/109
but when importing, there is an error: https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/110, which is actually ok, because there is an existing SPR e-item, and EBL e-item is not created, but this is confusing, because it seems there is an error.~~ comment: the difference is coming from the fact that "ignore missing rules" option was selected 
- Line 1: no error in preview: https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/123 but error on the e-item when importing: https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/124
- Preview https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/125 line 41 does not show an error but when importing https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/127 error because dependencies. 
~~https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/120 : line 21 no error but error when importing : https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/121~~ comment: there is a warning about duplicate series in the preview
=> It is important that there is a coherence between Preview and Import, otherwise we need to check twice. 

**Partial matches**
Partial matches: the workflow works as it is : from one test carried out: 697 records -> 15 new records, 3 correct matches (needs to be updated manually), it looks ok to handle this manually the correct matches (if the amount of records stays low).
However, would it be possible to know the criteria for partial matches? 
Some matches look strange: 
Ex in https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/109: 
- There is a partial matches although same ISBN: https://sandbox-cds-ils.web.cern.ch/backoffice/documents/qbtb2-ppr03 partial matches with: https://ebookcentral.proquest.com/lib/cern/detail.action?docID=6295127 (Line 175), the match should be automatic there. 
- Partial matches: some matches do not have the same authors (line 619)
https://ebookcentral.proquest.com/lib/cern/detail.action?docID=6388641 partial matches with:
https://sandbox-cds-ils.web.cern.ch/backoffice/documents/w6jxa-q1709 -> not same author
https://sandbox-cds-ils.web.cern.ch/backoffice/documents/fn64p-t8p52
https://sandbox-cds-ils.web.cern.ch/backoffice/documents/vchm4-cc737
https://sandbox-cds-ils.web.cern.ch/backoffice/documents/vrzfq-v6e17 -> not same author
https://sandbox-cds-ils.web.cern.ch/backoffice/documents/a89kb-fee06 -> not same author
https://sandbox-cds-ils.web.cern.ch/backoffice/documents/vchm4-cc737

**Series**
- Would it possible to by default remove " ser." at the end of the original series title when converting  and
- For the matching, ignore the values "ser." and "series" on both side (incoming records and ILS records). 
I.e. If a record arrives with value "Lecture notes in physics series", it should match with "Lecture notes in physics". 
E.g.:
- https://sandbox-cds-ils.web.cern.ch/backoffice/series/cfzdy-fqr42 duplicate created: https://sandbox-cds-ils.web.cern.ch/backoffice/series/m1gwx-cb757
- https://sandbox-cds-ils.web.cern.ch/backoffice/series/1vdyk-0pw94 / https://sandbox-cds-ils.web.cern.ch/backoffice/series/bb0s1-fag97 duplicate created: https://sandbox-cds-ils.web.cern.ch/backoffice/series/b0c4f-fjg72 

**Conversion general**
- [x] Date: do not add the date in Imprint date but only in Publication year
- [x] Author: add by default 'Person' when no other indication?
- [x] Author: add Roles according to the metadata, 'Author' by default when no other indication
- [ ] Language missing for Subtitle -> use same as for the document

**Springer import mode**
- [x] BUG https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/119 : line 98 and 238 wrong match -> 1st vs. 2nd ed. Should not update but do a partial match.
- [x] Remove "/" at the end of the title/subtitle (245__ a and 247__ b). Ex. "Conservative Classical and Quantum Systems /" -> "Conservative Classical and Quantum Systems". 
- [ ] Edition (250__ a) -> remove year ex. "3rd ed. 2021." -> "3rd"
- [x] Imprint place (264__ a) -> remove " :" at the end ex. "Cham :" -> "Cham"
- [ ] Add DOI in Identifiers scheme DOI (024_7_ a)
- [ ] ISBN -> add "Refers to material" -> "Digital version"
- [ ] ISBN -> add the ISBNs of the Print versions (776__ z) Identifiers, scheme: ISBN, material: "Print version"
- [x] Add Language:English
- [x] Remove ponctuation in series ISSN (490__ x) ex: "0168-1222 ;" -> "0168-1222"  

**EBL import mode**
- [ ] number of pages (300__a) not correctly extracted: "1 online resource (129 pages)" -> "129" and not "1"
- [x] EzProxy option should be selected in the e-item
- [ ] Ignore field 830
- [x] Ignore subfield 050_4$b

**EBL delete mode**

Failing files: 
- tried three times but did not manage to preview a file with 209 records. (EBL-exemple-del-1-failed.xml)
"Failed to import
The import of the literature failed, please try again.
If this error persists contact our technical support."

- worked then for another file with 388 records (EBL-exemple-del-2-ok.xml)
preview: https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/115
Line 5: there should be no partial matches for EBC deletions. In this case, the match is https://sandbox-cds-ils.web.cern.ch/backoffice/documents/ydrzy-ygx53 which has the e-item with the same URL and the same EBC identifier on the document. `comment: it matches partially by the ISBN`
import: https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/118
Line 2: no error shown in the preview, but error in the import. Document not updated (the EBC identifier should be removed). E-item well deleted but not showed in the import.
It shows errors when cannot delete the document because other dependencies. Ok for other e-items and items but maybe it would be good to easily distinguish the ones with other dependencies such as series/multiparts/editions, because the record can be deleted, e.g.: 
https://sandbox-cds-ils.web.cern.ch/backoffice/documents/wanh9-jn741 -> e-item deleted, document not deleted because link to other editions. Deletion to be done manually. (tracked in https://github.com/CERNDocumentServer/cds-ils/issues/608)

- then failed for another files with 422 records. (EBL-exemple-del-3-failed.xml)

**Safari import mode**
- [x] Title -> remove ":" at the end ex. "1,001 Ways to Make More Money as a Speaker, Consultant or Trainer: " -> 1,001 Ways to Make More Money as a Speaker, Consultant or Trainer"
- [x] Copyright: do not import
- [ ] Number of pages (300__a) not correctly extracted: "1 online resource (129 pages)" -> "129" and not "1"
- [x] EzProxy option should not be selected in the e-item
- [x] Convert <marc:datafield tag="100" ind1="#" ind2=" "> as if <marc:datafield tag="100" ind1=" " ind2=" ">
- [x] Convert <marc:datafield tag="245" ind1="1" ind2="2"> as if <marc:datafield tag="245" ind1="1" ind2="0">
- [x] Convert <marc:datafield tag="245" ind1="1" ind2="3"> as if <marc:datafield tag="245" ind1="1" ind2="0">
- [ ] In https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/123 line 1: https://sandbox-cds-ils.web.cern.ch/backoffice/documents/kpmc9-d4f88 matched with https://learning.oreilly.com/library/view/-/9781119691822/?ar which is not the same book (not same title, not same author) `comment: it has the same ISBN`

**Safari delete mode**

- [ ] https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/125
Line 118 : the Partial match should be an match exact (same URL on the e-item) `comment: different ISBNs`
~~https://sandbox-cds-ils.web.cern.ch/backoffice/importer/task/127 Line 294: e-item should have been deleted and therefore deletion of document too~~ **comment:** the e-item was not deleted, because it was never specified to be safari e-item (check the item of the document) therefore the script will not delete an eitem which does not belong to a correct provider

**For later**
~~Since there is now a field for the source at the e-item level, we will need at some point to convert the internal note that was used before at the importation (ex SAF202109) in the source field and creation dates.  (Only for e-items created before Septembre 2021).~~ **comment**: source field is already added

### PR DESCRIPTION
**Pull request by kprzerwa**
* importer: import fixes:

* closes #606
* closes #604

### CODE REVIEW

File: cds_ils/importer/documents/importer.py [hunk 1]
```diff
@@ -251,6 +252,10 @@ def search_for_matching_documents(self):
             results = search.scan()
             matches += [x.pid for x in results if x.pid not in matches]
 
+            for match in matches:
+                document = document_class.get_record_by_pid(match)
+                if document.get("edition") != self.json_data.get("edition"):
```
Comment (ntarocco) [inline cds_ils/importer/documents/importer.py:257]: if one have does not have the `edition` field, and the other has it, it will pass the test. Is it the expected behaviour?

File: cds_ils/importer/providers/ebl/importer.py [hunk 2]
```diff
@@ -14,4 +14,4 @@ class EBLImporter(Importer):
     """EBL importer class."""
 
     EITEM_OPEN_ACCESS = False
-    EITEM_URLS_LOGIN_REQUIRED = False
+    EITEM_URLS_LOGIN_REQUIRED = True
```
Comment (ntarocco) [inline cds_ils/importer/providers/ebl/importer.py:17]: Is this the default value? Does it mean that it was not correct before?

File: cds_ils/importer/providers/ebl/rules/document.py [hunk 3]
```diff
@@ -142,8 +144,13 @@ def alternative_identifiers(self, key, value):
 def languages(self, key, value):
     """Translates languages fields."""
     lang = clean_val("b", value, str).lower()
+    languages = self.get("languages")
```
Comment (ntarocco) [inline cds_ils/importer/providers/ebl/rules/document.py:147]: if the field is optional, you should have `[]` as default value:
```suggestion
    languages = self.get("languages", [])
```

File: cds_ils/importer/providers/safari/importer.py [hunk 4]
```diff
@@ -14,3 +14,4 @@ class SafariImporter(Importer):
     """OReilly importer class."""
 
     EITEM_OPEN_ACCESS = False
+    EITEM_URLS_LOGIN_REQUIRED = False
```
Comment (ntarocco) [inline cds_ils/importer/providers/safari/importer.py:17]: same question here

File: cds_ils/importer/views.py [hunk 5]
```diff
@@ -159,7 +159,7 @@ def post(self, **kwargs):
     @need_permissions("document-importer")
     def get(self):
         """Get method."""
-        list_count = 10
+        list_count = 50
```
Comment (ntarocco) [inline cds_ils/importer/views.py:162]: why this change?

File: cds_ils/importer/documents/importer.py (hunk 1 above)
Comment (kprzerwa) [reply cds_ils/importer/documents/importer.py:257]: yes, thats exactly what we want

File: cds_ils/importer/views.py (hunk 5 above)
Comment (kprzerwa) [reply cds_ils/importer/views.py:162]: requested in the ticket

File: cds_ils/importer/providers/ebl/importer.py (hunk 2 above)
Comment (kprzerwa) [reply cds_ils/importer/providers/ebl/importer.py:17]: its the default value, the library requested this change after testing, it seems it was wrong, same in the comment below

**Review by ntarocco — APPROVED** [review]

File: cds_ils/importer/providers/ebl/ignore_fields.py [hunk 6]
```diff
@@ -17,6 +17,7 @@
     "040__e",
     "040__c",
     "040__d",
+    "050_4b",
```
Comment (Pineirin) [inline cds_ils/importer/providers/ebl/ignore_fields.py:20]: What's the criteria behind ignoring some fields?

File: cds_ils/importer/providers/safari/rules/document.py [hunk 7]
```diff
@@ -34,7 +34,7 @@ def agency_code(self, key, value):
     return value
 
 
-@model.over("authors", "(^1001_)|(^7001_)")
+@model.over("authors", "(^1001_)|(^7001_)|(^100#_)|(^700#_)")
```
Comment (Pineirin) [inline cds_ils/importer/providers/safari/rules/document.py:37]: If the "#" a wildcard and if so what's the point of the first two options?

File: cds_ils/importer/providers/ebl/ignore_fields.py (hunk 6 above)
Comment (kprzerwa) [reply cds_ils/importer/providers/ebl/ignore_fields.py:20]: usually a request from the librarian

File: cds_ils/importer/providers/safari/rules/document.py (hunk 7 above)
Comment (kprzerwa) [reply cds_ils/importer/providers/safari/rules/document.py:37]: # is not a wildcard (usually `*` is a wildcard), but a MARC field indicator, see for example: https://cds.cern.ch/record/2782002/export/xm?ln=en (`ind1=", ind2="` etc)
